### ISSUE TITLE
linux-headers-surface

### ISSUE THREAD

**Issue by sourestdeeds** [linux-surface/linux-surface#364]
I get this far in the installation for ubuntu and its unable to locate linux-headers-surface. Where am i going wrong?

$ sudo apt install linux-image-surface linux-headers-surface iptsd libwacom-surface
$ sudo systemctl enable iptsd

**Comment by qzed** [comment]
Does it find `linux-image-surface` or any of the other packages? If not, did you add the package repository?

**Comment by sourestdeeds** [comment]
I did all the steps up to that point, let me try again and ill give a more concise output. Sorry, my request for help wasnt very informative!

**Comment by sourestdeeds** [comment]
I also get an error here saying the repository does not have a release file

$ echo "deb [arch=amd64] https://pkg.surfacelinux.com/debian release main" \
	| sudo tee /etc/apt/sources.list.d/linux-surface.list

$ sudo apt update

Am i just typing them in incorrectly?

**Comment by qzed** [comment]
I think you may also need to run `apt update` after adding the repo, so that all new metadata is fetched.

**Comment by sourestdeeds** [comment]
Yeah, i get stuck at the same point with linux headers, saying I have held broken packages. Perhaps i need to reinstall ubuntu?

**Comment by qzed** [comment]
I don't think you need to reinstall ubuntu. Although I have no clue what's going wrong (I don't know much about debian repositories unfortunately).

**Comment by sourestdeeds** [comment]
I think perhaps its conflicted with jakes install? Im not sure how to clean it all up to start again from scratch

**Comment by StollD** [comment]
I just ran all commands from the wiki page in a container and everything appears to work as intended.

Can you please post the output of all the commands you are running?

**Comment by sourestdeeds** [comment]
![Uploading Screenshot from 2020-12-19 22-31-37.png…]()

**Comment by sourestdeeds** [comment]
![Screenshot from 2020-12-19 22-33-07](https://user-images.githubusercontent.com/10788239/102700981-38081900-424a-11eb-8761-53405653711c.png)

**Comment by sourestdeeds** [comment]
![Screenshot from 2020-12-19 22-35-24](https://user-images.githubusercontent.com/10788239/102701001-861d1c80-424a-11eb-97e8-12b886386100.png)

**Comment by StollD** [comment]
```
The repository [some random repository thats definitly not from us] does not have a release file
```

You have an invalid repository added to APT. You need to remove that, otherwise APT wont be able to update the metadata from any repository, not just the failing one.

**Comment by sourestdeeds** [comment]
What's the best way to remove all the previous bits? Its a fresh install, but I tried using jakes and that didnt seem to work, but then found this.

**Comment by StollD** [comment]
This doesnt have anything to do with jakedays repo (jake never had a repo, just deb files on GitHub). You added a PPA for grub-customizer, and that PPA is broken. 

You need to remove that PPA from `/etc/apt/sources.list.d`, otherwise APT refuses to download the metadata for our repository.

**Comment by StollD** [comment]
Although, after looking at the screenshots more closely, this seems a bit confusing. Why can it resolve dependencies for linux-headers-surface, and why does that depend on linux-image-4.4-surface. Kernel 4.4 is like 4 years old by now.

But I dont really know anything about debian, so please try to fix the broken repo / apt update first. That seems like the most obvious issue.

**Comment by StollD** [comment]
Ok, now I see. You also added this repository which is just terribly out of date and most likely conflicts with ours. https://launchpad.net/~tigerite/+archive/ubuntu/kernel

Please remove that too.

**Comment by sourestdeeds** [comment]
This is my first time using ubuntu, how do I easily remove those? My old laptop broke and I really dislike windows, I was using mac before. Sorry if I sound like a total newbie

**Comment by sourestdeeds** [comment]
I've figured it out, trying now with them removed

**Comment by sourestdeeds** [comment]
So i ran all the installation steps, but the linux-surface fails to boot, and says i need to load the kernel first?

**Comment by qzed** [comment]
> So i ran all the installation steps, but the linux-surface fails to boot, and says i need to load the kernel first?

That sounds like secure-boot isn't set up for the linux-surface kernel. To set that up, you'll have to boot into the normal kernel, install the `linux-surface-secureboot-mok` package and follow the instructions printed at installation, i.e. you'll have to reboot and add the secureboot key via the MokManager tool.

**Comment by sourestdeeds** [comment]
I think i need to reinstall it all as I saw that pop up but left if for 10 seconds without doing anything. Oops

**Comment by sourestdeeds** [comment]
Darn, its not popping up again when I restart. I've really messed this up. Just too stressed as I lost half a terms work when my old laptop died. Making stupid mistakes :(

**Comment by sourestdeeds** [comment]
Ok, uninstalled mok and installed, rebooted and it works!!!

**Comment by qzed** [comment]
If you've missed the instructions for the `linux-surface-secureboot-mok` package, you can also just uninstall that (probably reboot in between) and then re-install it. The instruction should then show up again.

**Comment by qzed** [comment]
Neat!

**Comment by sourestdeeds** [comment]
I'm so grateful for your patience in helping me. You have no idea how much you've saved my 2020!

**Comment by qzed** [comment]
Happy to help.
